compiler/src/abi/builder.rs

```diff
@@ -116,10 +116,14 @@ fn func_def<'a>(
         .collect::<Result<Vec<FuncInput>, CompileError>>()?;
 
     let outputs = if let Some(return_type) = return_type {
-        vec![FuncOutput {
-            name: "".to_string(),
-            typ: type_desc(type_defs, &return_type.node)?,
-        }]
+        match type_desc(type_defs, &return_type.node)? {
+            // Should there be a different ABI for `pub def foo() -> ():` and `pub def foo():`?
```

Comment:
@g-r-a-n-t I'm unsure about this. On the one hand, it would feel weird if `pub def foo() -> ():` and `pub def foo():` resulted in different ABI types. On the other hand, **forcing**  `pub def foo() -> (): to become `pub def foo():` would prevent people from matching a given ABI spec they are trying to implement. Or is this a non-issue because of something that I might be missing?

Reply:
I think we want to treat `pub def foo() -> ():` the same as `pub def foo():`, since both technically return an empty tuple. It might even make sense to emit a warning in the case of `pub def foo() -> ():`.

The ABI spec does say that you can have an empty tuple..

> [`(T1,...,Tk)` for `k >= 0` and any types `T1, …, Tk`](https://docs.soliditylang.org/en/v0.8.1/abi-spec.html#formal-specification-of-the-encoding)

So there is room for this sort of thing in the ABI, I'm just very skeptical of there being an instance where putting an empty tuple in the json ABI makes sense.

Reply:
>I'm just very skeptical of there being an instance where putting an empty tuple in the json ABI makes sense

So am I but let's say there's a project that defines some interface `IStrategy` with a method `pub def do_something(val: Whatever) -> ()`. The way I understand it, one would not be able to write a strategy in Fe that produces an ABI that matches that ABI signature because the Fe compiler would always normalize it to `pub def do_something(val: Whatever):`

Reply:
Ok, I'll merge this as is. If this turns out to become a problem we can address it later.